src/views/Governance/components/DelegateModal/DelegateModal.tsx

```diff
@@ -19,6 +20,23 @@ If votes aren't delegated anywhere this input should either be empty or it shoul
  */
 const DelegateModal: FC<DelegateModalProps> = ({ open, handleClose }) => {
   const theme = useTheme();
+  const [delegatedAddress, setDelegatedAddress] = useState("");
+  const [isDelegatedAddressValid, setIsDelegatedAddressValid] = useState(true);
+
+  const handleSetDelegatedAddress = (value: string) => {
+    setDelegatedAddress(value);
+    checkIsAddressValid(value);
+  };
+
+  const checkIsAddressValid = (value: string) => {
+    if (!isAddress(value)) {
+      setIsDelegatedAddressValid(false);
+      return;
+    }
+
```

Comment:
we have a helper function in src/helpers/misc  called isValidAddress , that we may want to use instead here instead. It also covers the AddressZero case.

Reply:
switched